#!/usr/bin/env python3

from jinja2 import Template


include: "rules/common.smk"
include: "rules/samtools_import.smk"
include: "rules/samtools_fasta.smk"


#############
# FUNCTIONS #
#############



########
# MAIN #
########


# this is from the bpa_dataportal_downloads project
# results of a search for query = {"sample_id": '102.100.100/411655'}
data_file_dict = {
    "414129_AusARG_AGRF_DA235386.subreads.bam": "https://data.bioplatforms.com/dataset/bpa-ausarg-pacbio-hifi-414129-da235386/resource/f28fe709744235efc0da895975c68f6f/download/414129_AusARG_AGRF_DA235386.subreads.bam",
    "414129_AusARG_AGRF_DA235386.ccs.bam": "https://data.bioplatforms.com/dataset/bpa-ausarg-pacbio-hifi-414129-da235386/resource/27fc8f6c8fb9e3b462a3cdae002bd5f4/download/414129_AusARG_AGRF_DA235386.ccs.bam",
    "414129_AusARG_AGRF_DA235337.subreads.bam": "https://data.bioplatforms.com/dataset/bpa-ausarg-pacbio-hifi-414129-da235337/resource/303068e96c58206aafd222a2cad04545/download/414129_AusARG_AGRF_DA235337.subreads.bam",
    "414129_AusARG_AGRF_DA235337.ccs.bam": "https://data.bioplatforms.com/dataset/bpa-ausarg-pacbio-hifi-414129-da235337/resource/bdd8b8ba65c217945353138c1ba0d4be/download/414129_AusARG_AGRF_DA235337.ccs.bam",
    "414130_AusARG_BRF_HKWJJDMXY_AAGCATCG_S5_R2_001.fastq.gz": "https://data.bioplatforms.com/dataset/bpa-ausarg-hi-c-414130-hkwjjdmxy/resource/71e2e0c5384d238811bd78e54cbe0111/download/414130_AusARG_BRF_HKWJJDMXY_AAGCATCG_S5_R2_001.fastq.gz",
    "414130_AusARG_BRF_HKWJJDMXY_AAGCATCG_S5_R1_001.fastq.gz": "https://data.bioplatforms.com/dataset/bpa-ausarg-hi-c-414130-hkwjjdmxy/resource/796d9cd507b33eb34b3b31e4200129c7/download/414130_AusARG_BRF_HKWJJDMXY_AAGCATCG_S5_R1_001.fastq.gz",
}


#########
# RULES #
#########

# TARGET IS AT THE END


# # Run the pipeline
# rule sanger_tol_genomeassembly:
#     input:
#         input=Path(outdir, "config", "sangertol_genomeassembly_params.yaml"),
#     output:
#         report="results/multiqc/broadPeak/multiqc_report.html",
#     params:
#         pipeline="sanger-tol/genomeassembly",
#         revision="0.10.0",
#         profile=["singularity", "spartan"],
#         outdir=lambda wildcards, output: str(Path(output.report).parents[-2]),
#     handover: True
#     wrapper:
#         "v5.5.0-8-gb18ec90/utils/nextflow"


# Set up the pipeline






###########
# TARGETS #
###########


rule format_config_file:
    input:
        sanger_config_template=local(sanger_config_template),
    output:
        rendered_yaml=add_bucket_to_path(
            Path(
                dataset_id, "results", "config", "sangertol_genomeassembly_params.yaml"
            )
        ),
    params:
        pacbio_reads=[rules.samtools_fasta.output.reads],
        hic_reads=[rules.samtools_import.output.cram],
    container:
        get_container("ncbi-datasets-pylib")
    script:
        "scripts/format_config_file.py"
